tst/EnergyPlus/unit/FuelCellElectricGenerator.unit.cc

```diff
@@ -55,132 +55,1027 @@
+    auto thisFC = dynamic_cast<FuelCellElectricGenerator::FCDataStruct*> (thisFCcompPtr);
+    // Power Module
+    EXPECT_EQ("GENERATOR FUEL CELL POWER MODULE 1", thisFC->NameFCPM);
+    auto fCPM = thisFC->FCPM;
+    EXPECT_EQ(fCPM.Name, thisFC->NameFCPM);
+    // Annex42 = Direct
+    EXPECT_EQ(DataGenerators::CurveMode::Direct, fCPM.EffMode);
+    ASSERT_GT(fCPM.EffCurveID, 0);
+    EXPECT_EQ("POWER MODULE EFFICIENCY CURVE", state->dataCurveManager->PerfCurve(fCPM.EffCurveID).Name);
+    EXPECT_EQ(1.0, fCPM.NomEff);
+    EXPECT_EQ(3400.0, fCPM.NomPel);
+    // At beggining of simulation, 10. But then it was simulated, so it should be greater
+    EXPECT_GT(fCPM.NumCycles, 10);
+    EXPECT_EQ(0.0, fCPM.CyclingDegradRat);
+    EXPECT_EQ(1.0, fCPM.NomEff);
+    EXPECT_EQ(3400.0, fCPM.NomPel);
+    EXPECT_EQ(0.0, fCPM.NumCycles);
+    EXPECT_EQ(0.0, fCPM.CyclingDegradRat);
+    EXPECT_EQ(0.0, fCPM.NumRunHours);
+    EXPECT_EQ(0.0, fCPM.OperateDegradRat);
+    EXPECT_EQ(10000.0, fCPM.ThreshRunHours);
+    EXPECT_EQ(1.4, fCPM.UpTranLimit);
+    EXPECT_EQ(0.2, fCPM.DownTranLimit);
+    EXPECT_EQ(0.0, fCPM.StartUpTime);
+    EXPECT_EQ(0.2, fCPM.StartUpFuel);
+    EXPECT_EQ(0.0, fCPM.StartUpElectConsum);
+    EXPECT_EQ(0.0, fCPM.StartUpElectProd);
+    EXPECT_EQ(0.0, fCPM.ShutDownTime);
+    EXPECT_EQ(0.2, fCPM.ShutDownFuel);
+    EXPECT_EQ(0.0, fCPM.ShutDownElectConsum);
+    EXPECT_EQ(0.0, fCPM.ANC0);
+    EXPECT_EQ(0.0, fCPM.ANC1);
+
+    EXPECT_EQ(DataGenerators::SkinLoss::ConstantRate, fCPM.SkinLossMode);
+
+    EXPECT_EQ("THERMAL ZONE 1", fCPM.ZoneName);
+    EXPECT_EQ(1, fCPM.ZoneID);
+
+    EXPECT_EQ(0.6392, fCPM.RadiativeFract);
+
+    // TODO: this is an actual bug. It is being overriden in FigureFuelCellZoneGains, while here the SkinLossMode is ConstantRate, so it should be
+    // unaffected throughout
+    EXPECT_EQ(729, fCPM.QdotSkin);
+
+
+    EXPECT_EQ(0.0, fCPM.UAskin);
+    EXPECT_GT(fCPM.SkinLossCurveID, 0);
+    EXPECT_EQ(0.006156, fCPM.NdotDilutionAir);
+    EXPECT_EQ(2307, fCPM.StackHeatLossToDilution);
+    EXPECT_EQ("GENERATOR FUEL CELL POWER MODULE 1 OA NODE", fCPM.DilutionInletNodeName);
+    EXPECT_EQ("GENERATOR FUEL CELL POWER MODULE 1 DILUTION OUTLET AIR NODE", fCPM.DilutionExhaustNodeName);
+    EXPECT_EQ(3010, fCPM.PelMin);
+    EXPECT_EQ(3728, fCPM.PelMax);
+
+
+
+    // Air Supply
+    EXPECT_EQ("GENERATOR FUEL CELL AIR SUPPLY 1", thisFC->NameFCAirSup);
+    auto airSup = thisFC->AirSup;
+    EXPECT_EQ(airSup.Name, thisFC->NameFCAirSup);
+    EXPECT_EQ("GENERATOR FUEL CELL AIR SUPPLY 1 OA NODE", airSup.NodeName);
+    ASSERT_GT(airSup.BlowerPowerCurveID, 0);
+    EXPECT_EQ("BLOWER POWER CURVE", state->dataCurveManager->PerfCurve(airSup.BlowerPowerCurveID).Name);
+    EXPECT_EQ(1.0, airSup.BlowerHeatLossFactor);
+    EXPECT_EQ(DataGenerators::AirSupRateMode::ConstantStoicsAirRat, airSup.AirSupRateMode);
+
+    // TODO: Interestingly, Stoics ratio is the input + 1.0
+    EXPECT_EQ(2.0, airSup.Stoics);
```

Comment:
haven't really had time to track how that makes sense, but will do.

Reply:
Well, at least the IO ref mentions it, so no problem: https://bigladdersoftware.com/epx/docs/9-4/input-output-reference/group-electric-load-center-generator.html#field-stoichiometric-ratio